src/ChargeTrapping/ChargeTrapping.jl

```diff
@@ -314,22 +324,22 @@ function _calculate_signal(
         if in_inactive_region
             if inactive_layer_ctmodel == "ConstantLifetime"
                 # Constant trapping lifetime in inactive layer
-                tmp_signal[i] = _constantlifetime_signal!( running_sum, τ_inactive, Δt, q, w )
+                tmp_signal[i] = _signal!( ctm.inactive_charge_trapping_model, q, w; running_sum=running_sum, τ=τ_inactive, Δt=Δt )
             else
                 # No charge trapping in inactive layer
-                tmp_signal[i] = _nochargetrapping_signal!(w, charge)
+                tmp_signal[i] = _signal!( ctm.inactive_charge_trapping_model, q, w )
             end
 
         else
             if bulk_ctmodel == "ConstantLifetime"
                 # Constant trapping lifetime in bulk
-                tmp_signal[i] = _constantlifetime_signal!( running_sum, τ, Δt, q, w )
+                tmp_signal[i] = _signal!( ctm.bulk_charge_trapping_model, q, w; running_sum=running_sum, τ=τ, Δt=Δt )
             elseif bulk_ctmodel == "Boggs"
                 # Boggs trapping model in bulk
-                tmp_signal[i] = _Boggs_signal!( running_sum, nσ, Δl, q, w )
+                tmp_signal[i] = _signal!( ctm.bulk_charge_trapping_model, q, w; running_sum=running_sum, nσ=nσ, Δl=Δl )
             else
                 # No charge trapping in bulk
-                tmp_signal[i] = _nochargetrapping_signal!(w, charge)
+                tmp_signal[i] = _signal!( ctm.bulk_charge_trapping_model, q, w )
             end
         end
     end
```

Comment:
Do you need these `if`-`else` statements at all now?
My idea behind `_signal!` was to be able to replace this whole thing with 
```julia
if in_inactive_region
    tmp_signal[i] = _signal!(ctm.inactive_charge_trapping_model, q, w; running_sum, τ = τ_inactive, Δt)
else
    tmp_signal[i] = _signal!(ctm.bulk_charge_trapping_model, q, w; running_sum, τ, Δt, nσ, Δl)
end
```
(No need to check for `bulk_ctmodel` or `inactive_layer_ctmodel` anymore, because that is passed to `_signal!` now.